src/pretix/static/pretixbase/js/addressform.js

```diff
@@ -39,7 +40,7 @@ $(function () {
                             required = 'required' in options && options.required && isRequired && visible;
 
                         dependent.closest(".form-group").toggle(visible).toggleClass('required', required);
-                        dependent.prop("required", required);
+                        dependent.prop("required", required && htmlRequired);
```

Comment:
Especially used in this context IMHO naming it `forceOptional` is easier to read/understand for my brain.

```suggestion
                        dependent.prop("required", required && !forceOptional);
```